.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -29,12 +29,86 @@ pull_test_dockers_arm64:
   variables:
     ARCH: arm64
 
+.wait_for_instances:
+  - |
+    COUNTER=0
+    while [[ $(aws ec2 describe-instances --filters 'Name=tag:team,Values=ebpf-platform' 'Name=tag:managed-by,Values=pulumi' 'Name=instance-state-name,Values=running' "Name=tag:pipeline-id,Values=${CI_PIPELINE_ID}" --output text --query 'Reservations[*].Instances[*].InstanceId' | wc -l ) != "2" && $COUNTER -le 40 ]]; do COUNTER=$[$COUNTER +1]; echo "[${COUNTER}] Waiting for instances"; sleep 30; done
+    # check that 2 instances are ready, or fail
+    if [ $(aws ec2 describe-instances --filters 'Name=tag:team,Values=ebpf-platform' 'Name=tag:managed-by,Values=pulumi' 'Name=instance-state-name,Values=running' "Name=tag:pipeline-id,Values=${CI_PIPELINE_ID}" --output text --query 'Reservations[*].Instances[*].InstanceId' | wc -l) -ne "2" ]; then
+        echo "Both instances NOT found"
+        "false"
```

Comment:
What does this do?

Reply:
`false` runs `/bin/false` which return a non-zero exit code, causing the job to fail.

Reply:
Why is it weirdly quoted?

Reply:
Because it is a keyword. Gitlab gets confused otherwise. The quotes force it to interpret it as a shell command.